runtime/compiler/x/codegen/J9CodeGenerator.cpp

```diff
@@ -383,7 +390,6 @@ J9::X86::CodeGenerator::suppressInliningOfRecognizedMethod(TR::RecognizedMethod
    switch (method)
       {
       case TR::java_lang_Object_clone:
-      case TR::com_ibm_jit_JITHelpers_findElementFromArray:
```

Comment:
Can you describe why this line is deleted? IIRC we suppress inlining of recognized methods in `supressInliningRecognizedInitialCallee` in the inliner, but seems we also have a codegen query. This should perhaps be consolidated into one list. Which one are we supposed to use? @vijaysun-omr @0dvictor thoughts?

Reply:
I thought that suppressing twice in `supressInliningRecognizedInitialCallee` and inside the platform specific code generators was redundant. It seems like many other methods like `String.hashCode` and `String.toUpper` were found there and not in `suppressInliningOfRecognizedMethod` so I though moving this method there instead of here was appropriate.

Reply:
Either place can prevent optimizer from inlining the method, and we should only put the method in one of the lists. I personally think it fits more in `CodeGenerator::suppressInliningOfRecognizedMethod()` because it is the CodeGen who requests Optimizer to suppress inlining of this method. Unfortunately, there is not a unified place to suppress inlining in all the platforms. `TR_J9VMBase::supressInliningRecognizedInitialCallee()` seems less relevant to CodeGen and it sounds more like doing certain work for ILGen or Optimizer; however, its many entries are actually for CodeGen and is used as the consolidated place for suppress inlining of methods on multiple platforms.
therefor, for this specific pull request, I agree with @dhong44 to move `com_ibm_jit_JITHelpers_findElementFromArray` up so that multiple platforms can share the same code. In the long run, I second @fjeremic 's idea about consolidate `CodeGenerator::suppressInliningOfRecognizedMethod`. It may be as simple as a default implementation of `J9::CodeGenerator` and each platform's CodeGen calls the parent before checking its own list.